nix/modules/flake-parts/lib.nix

```
@@ -1,8 +1,8 @@
-{ self, inputs, config, lib, ... }:
+{ self, inputs, config, lib, specialArgs, ... }:
 let
   specialArgsFor = rec {
     common = {
-      flake = { inherit self inputs config; };
+      flake = { inherit self inputs config specialArgs; } // specialArgs;
```

Let's avoice redundancy. And use it in a single place uniformly.

```suggestion
      flake = { inherit self inputs config; } // specialArgs;
```

Also, is `//` operator appropriate here? What happens if the user overrides pre-defined args like `self`, `inputs`, `config` (accidentally or otherwise)?

---

Hmm, I am quite new to nix, so I don't have a better suggestion. 

Based on my experience, clobbering `config` also happens with `home-manager` if you accidentally passed in the `config` from `nix-darwin` in `specialArgs`. So I suppose this is a behavior the user should be aware of. If they passed in `self` and `config` in `specialArgs`, it would also have clobbered the arguments to the `flake-parts` module because `specialArgs` is merged into the arguments this way too.

https://github.com/NixOS/nixpkgs/blob/3c9c51328c654a95c942b6b4105182a3a9c372b0/lib/modules.nix#L240-L246

Based on my brief understanding, `flake-parts`, `nix-darwin` and `home-manager` passes `specialArgs` to `lib.evalModules` under the hood and it is merged into arguments this way. So I'd say this is the expected behaviour.